src/Build.UnitTests/BackEnd/BuildManager_Tests.cs

```diff
@@ -1517,7 +1517,8 @@ public void CancelledBuild()
 
             asyncResult.ExecuteAsync(null, null);
             _buildManager.CancelAllSubmissions();
-            asyncResult.WaitHandle.WaitOne();
+            asyncResult.WaitHandle.WaitOne(10000);
```

Comment:
```suggestion
            asyncResult.WaitHandle.WaitOne(TimeSpan.FromSeconds(10));
```

I always prefer the explicit overload, so you don't have to remember units.